deploy-service/common/src/main/java/com/pinterest/deployservice/rodimus/RodimusManagerImpl.java

```diff
@@ -68,31 +73,97 @@ public boolean shouldSkipClass(Class<?> c) {
         }
     }
 
+    private boolean refreshCachedKey() throws Exception {
+        String prevKnoxKey = this.catchedKey;
+
+        if (this.fsKnox != null) {
+            this.catchedKey = new String(this.fsKnox.getPrimaryKey());
+        }
+
+        if ( prevKnoxKey == null )
+        {
+            return this.catchedKey != null;
+        }else{
+            return ! prevKnoxKey.equals( this.catchedKey );
```

Comment:
should this return false if refresh was successful ?

How this return value going to be used?

Reply:
if refresh is successful (new key not equal previous one) then return true.

example:
               ` if ( ! this.refreshCachedKey() ) throw e;`